app/controllers/products_controller.rb

```diff
@@ -70,9 +67,41 @@ def update
   end
 
   def new
+    unless session[:merchant_id]
+      flash[:status] = :error
+      flash[:result_text] = "You need to be logged in to create a product!"
+      redirect_back fallback_location: root_path, status: 403
+    end
+
+    @product = Product.new(merchant_id: session[:merchant_id])
+    @categories = Product.categories
   end
 
   def create
+    @product = Product.new(product_params)
+
+    if params[:categories]
```

Comment:
can you use the Product instance method update_categories here like you did in the update method (line 54)?